src/main.rs

```diff
@@ -330,56 +334,86 @@ async fn transactions(
     // Initialize the result variables
     let mut transaction_results: TransactionResult = TransactionResult::default();
 
-    // Return the combined results
+    let mut eth_send_query = ethereum_sends.into_boxed();
     if let Some(eth_address) = address_query.eth_address {
-        let ethereum_sends_results = ethereum_sends
-            .select(EthereumSend::as_select())
-            .filter(schema::ethereum_sends::depositor_address.eq(format!("{:?}", eth_address)))
-            .order_by(schema::ethereum_sends::source_timestamp.desc())
-            .limit(500)
-            .load::<EthereumSend>(&mut conn);
-
-        match ethereum_sends_results {
-            Ok(transaction) => {
-                transaction_results.eth_send = transaction
-                    .into_iter()
-                    .map(map_ethereum_send_to_transaction_result)
-                    .collect()
-            }
-            Err(e) => {
-                tracing::error!("Cannot get ethereum send transactions: {:?}", e);
-                return (
-                    StatusCode::INTERNAL_SERVER_ERROR,
-                    [("Cache-Control", "max-age=60, must-revalidate".to_string())],
-                    Json(json!({ "error": e.to_string()})),
-                );
-            }
+        eth_send_query = eth_send_query
+            .filter(schema::ethereum_sends::depositor_address.eq(format!("{:?}", eth_address)));
+    }
+
+    if let Some(avail_address) = address_query.avail_address {
+        eth_send_query = eth_send_query
+            .or_filter(schema::ethereum_sends::receiver_address.eq(format!("{:?}", avail_address)));
+    }
+
+    // Return the combined results
+    let ethereum_sends_results = eth_send_query
+        .select(EthereumSend::as_select())
+        .order_by(schema::ethereum_sends::source_timestamp.desc())
+        .limit(state.transactions_result_max_size.into())
```

Comment:
Please implement pagination with `offset` and integrate with endpoint.

Reply:
This simplifies ui